packages/x-date-pickers/src/TimeClock/TimeClock.tsx

```diff
@@ -263,8 +264,29 @@ export const TimeClock = React.forwardRef(function TimeClock<TDate extends unkno
           return false;
         }
 
+        if (shouldDisableClock) {
```

Comment:
We don't want to support both at the same time ?
Otherwise we need

```suggestion
        if (shouldDisableClock?.(timeValue, viewType)) {
          return false;
        }
```

Reply:
thank you for spotting it. 🤦 
[Fixed](https://github.com/mui/mui-x/pull/7299/commits/1b676926187fb68808712bc8011cb67ca21e9e30).